lib/os/shutdown.c

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/shutdown_internal.h>
+#include <zephyr/toolchain.h>
+
+#if defined(CONFIG_MULTITHREADING) && !defined(CONFIG_SMP)
+#include <zephyr/kernel.h>
+#endif
+
+void sys_shutdown(void)
+{
+#if defined(CONFIG_MULTITHREADING) && !defined(CONFIG_SMP)
+	z_sys_shutdown_schedule();
+	k_sleep(K_FOREVER);
+#else
+	z_sys_shutdown();
```

Comment:
This (in the platform implements I see here on NXP and Nordic) is synchronous, and will effect a shutdown immediately regardless of whatever the rest of the system is doing.